### PR TITLE
Adding new supported states to Ios_lldp_global resource module

### PR DESCRIPTION
**Pull request by justjais**
##### SUMMARY

Adding new supported states to Ios_lldp_global resource module

##### ISSUE TYPE

- Feature Pull Reques

##### COMPONENT NAME

ios_lldp_global

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: plugins/module_utils/network/ios/config/lldp_global/lldp_global.py [hunk 1]
```diff
@@ -83,20 +83,40 @@ def execute_module(self):
         commands = list()
         warnings = list()
 
-        existing_lldp_global_facts = self.get_lldp_global_facts()
-        commands.extend(self.set_config(existing_lldp_global_facts))
+        if self.state in self.ACTION_STATES:
+            existing_lldp_global_facts = self.get_lldp_global_facts()
+        else:
+            existing_lldp_global_facts = []
```
Comment (rohitthakur2590) [inline plugins/module_utils/network/ios/config/lldp_global/lldp_global.py:89]: this should be dict.

File: plugins/module_utils/network/ios/config/lldp_global/lldp_global.py [hunk 2]
```diff
@@ -83,20 +83,40 @@ def execute_module(self):
...
+            result["rendered"] = commands
+        elif self.state == "parsed":
+            running_config = self._module.params["running_config"]
+            if not running_config:
+                self._module.fail_json(
+                    msg="value of running_config parameter must not be empty for state parsed"
+                )
+            result["parsed"] = self.get_lldp_global_facts(data=running_config)
+        else:
+            changed_lldp_global_facts = []
```
Comment (rohitthakur2590) [inline plugins/module_utils/network/ios/config/lldp_global/lldp_global.py:112]: IMO it should be dict

File: plugins/modules/ios_lldp_global.py [hunk 3]
```diff
@@ -109,6 +109,14 @@
             description:
             - System Name TLV
             type: bool
+  running_config:
```
Comment (rohitthakur2590) [inline plugins/modules/ios_lldp_global.py:112]: In short description isn't it should be LLDP global resource module?

File: plugins/module_utils/network/ios/config/lldp_global/lldp_global.py [hunk 4]
```diff
@@ -124,21 +144,23 @@ def set_state(self, want, have):
                   to the deisred configuration
         """
         commands = []
-        state = self._module.params["state"]
-        if state in ("merged", "replaced") and not want:
+        if (
+            self.state in ("overridden", "merged", "replaced", "rendered")
```
Comment (rohitthakur2590) [inline plugins/module_utils/network/ios/config/lldp_global/lldp_global.py:148]: overridden not needed

File: plugins/module_utils/network/ios/argspec/lldp_global/lldp_global.py [hunk 5]
```diff
@@ -58,8 +58,17 @@ def __init__(self, **kwargs):
             },
             "type": "dict",
         },
+        "running_config": {"type": "str"},
         "state": {
-            "choices": ["merged", "replaced", "deleted"],
+            "choices": [
+                "merged",
+                "replaced",
+                "overridden",
```
Comment (rohitthakur2590) [inline plugins/module_utils/network/ios/argspec/lldp_global/lldp_global.py:66]: overridden not needed

**Comment by justjais** [comment]
@rohitthakur2590 thanks for the review, and have implemented the suggested changes.

**Comment by justjais** [comment]
recheck

**Comment by justjais** [comment]
recheck

**Pull request merged**